nvflare/app_common/workflows/scatter_and_gather.py

```diff
@@ -171,6 +171,14 @@ def start_controller(self, fl_ctx: FLContext) -> None:
         fl_ctx.set_prop(AppConstants.START_ROUND, self._start_round, private=True, sticky=True)
         fl_ctx.set_prop(AppConstants.NUM_ROUNDS, self._num_rounds, private=True, sticky=False)
         self._global_weights = self.persistor.load(fl_ctx)
+        if self._global_weights.is_empty():
+            # see whether it is available from fl_ctx
+            self._global_weights = fl_ctx.get_prop(AppConstants.GLOBAL_MODEL)
+
+        if self._global_weights.is_empty():
```

Comment:
this should change to `if self._global_weights is None`

Reply:
Or if we are forcing the `fl_ctx.get_prop(AppConstants.GLOBAL_MODEL)` to return an object `ModelShareable`, we need to check for the type before we call this `is_empty()` method

Reply:
Yes, we are returning `ModelLearnable` here. I added the type-checking in SAG.

Reply:
We need to check for the type of `fl_ctx.get_prop(AppConstants.GLOBAL_MODEL)`

Because it might get None or other types?

Reply:
It's still there in my branch https://github.com/holgerroth/NVFlare/blob/5083445861e2ecd101976860ff18307de55ab868/nvflare/app_common/workflows/scatter_and_gather.py#L175

Reply:
thanks for the confirm, my github was not showing, now I see it